### DISCUSSION TITLE
request http://xxx:12800/v3/segment  errorcode:405

### DISCUSSION THREAD

**Discussion by huangyedie** [apache/skywalking#10943] [Q&A]
Requesting the OAP interface to write data on its own and reporting an error

requestBody:{"TraceId":"a12ff60b-5807-463b-a1f8-fb1c8608219e","TraceSegmentId":"a12ff60b-5807-463b-a1f8-fb1c8608219e","Spans":[{"SpanId":1,"ParentSpanId":0,"StartTime":638223951265028760,"EndTime":638223951265029357,"Refs":[],"OperationName":"OperationName2","Peer":"","SpanType":2,"SpanLayer":1,"ComponentId":0,"IsError":false,"Tags":[],"Logs":[],"SkipAnalysis":false},{"SpanId":0,"ParentSpanId":-1,"StartTime":638223951265030800,"EndTime":638223951265030803,"Refs":[],"OperationName":"OperationName1","Peer":"","SpanType":2,"SpanLayer":1,"ComponentId":0,"IsError":false,"Tags":[],"Logs":[],"SkipAnalysis":false}],"Service":"test","ServiceInstance":"ServiceInstance","IsSizeLimited":false}

Can't I request it directly?

reference：
https://skywalking.apache.org/docs/main/next/en/api/trace-data-protocol-v3/#post-httplocalhost12800v3segment

**Comment by wu-sheng** [comment]
The rest API is open for reporting. Wha is your error? Have you debugged to see why?

**Comment by huangyedie** [comment]
response：
{Method: POST, RequestUri: 'http://xxxx:12800/v3/segment', Version: 1.1, Content: System.Net.Http.ByteArrayContent, Headers:
{
  Content-Type: application/json
  Content-Length: 678
}}

![image](https://github.com/apache/skywalking/assets/51399090/ee204d63-6f39-43bb-9959-86d9aef294ed)

**Reply by wu-sheng** [reply]
Check `TraceSegmentReportHandler#collectSegments` of OAP, you could debug from source codes. If you can't get the idea.
But, from my guess, you either has the wrong IP binding/being-used or some local env issue.
`405` should haven't relationship with the code logic.

**Reply by wu-sheng** [reply]
Either way, it is hard to tell from our side. I can only say, this URI working in our end.

**Reply by huangyedie** [reply]
Okay, let me check the local env issue

**Comment by kezhenxu94** [comment]
this API only support `POST` method, and you are sending `GET` request, `405` is a basic http error code

**Comment by huangyedie** [comment]
> this API only support `POST` method, and you are sending `GET` request, `405` is a basic http error code

I am using a post request

**Comment by huangyedie** [comment]
![image](https://github.com/apache/skywalking/assets/51399090/fefff7f4-8217-4982-b0d5-0cf04f0841ab)

"HTTP method POST is not supported by this URL"
http://xxx:12800/v3/segment
Is there a problem with this?

**Comment by huangyedie** [comment]
I will prompt 404 when requesting with get, and 405 when using post

**Comment by huangyedie** [comment]
Problem solving: OAP version issue
